pymeasure/adapters/serial.py

```diff
@@ -45,7 +45,7 @@ class SerialAdapter(Adapter):
 
     def __init__(self, port, preprocess_reply=None, **kwargs):
         super().__init__(preprocess_reply=preprocess_reply)
-        if isinstance(port, serial.Serial):
+        if isinstance(port, serial.SerialBase):
```

Comment:
I'm curious why this change was needed -- afaict, `serial_for_url` [returns a `Serial` instance](https://github.com/pyserial/pyserial/blob/0e7634747568547b8a7f9fd0c48ed74f16af4b23/serial/__init__.py#L61), not `SerialBase`.

Reply:
@bilderbuchi , here what I found in my ubuntu machine:
```
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import serial
>>> serial.__version__
'3.4'
>>> s=serial.serial_for_url("loop://")
>>> s
Serial<id=0x7fe03456bcd0, open=True>(port='loop://', baudrate=9600, bytesize=8, parity='N', stopbits=1, timeout=None, xonxoff=False, rtscts=False, dsrdtr=False)
>>> s.__class__
<class 'serial.urlhandler.protocol_loop.Serial'>
>>> s.__class__.__base__
<class 'serial.serialutil.SerialBase'>
>>> s.__class__.__mro__
(<class 'serial.urlhandler.protocol_loop.Serial'>, <class 'serial.serialutil.SerialBase'>, <class 'io.RawIOBase'>, <class '_io._RawIOBase'>, <class 'io.IOBase'>, <class '_io._IOBase'>, <class 'object'>)
>>> isinstance(s, serial.Serial)
False
```
As you can see, loopback class is not an instance of `serial.Serial`, at least in the version I have.

Reply:
Thank you for the analysis, I can confirm; this is fine I think. Pyserial chose an interesting design with several *different* classes called `Serial` :grin: .